leader/leader.go

```diff
@@ -99,32 +99,33 @@ type CallbackFn func(key, prevValue, newValue string)
 func (l *Client) AddWatchCallback(key string, fn CallbackFn) {
 	ctx, cancel := context.WithCancel(context.Background())
 	valuesC := make(chan string)
+	l.wg.Add(3)
```

Comment:
I'd maybe at minimum add a comment on where the 3 count gets cancelled, as looking at just the function make this look like a bug as we add 3 but in this function only cancel twice unless I'm missing it. 

Either that or add the third add where the goroutine is launched so it's less likely to mistakenly get "fixed".

Reply:
Fair point - the "invisible" `wg.Add` is in the `addWatch` - I'll make it more explicit.